singer_sdk/streams/core.py

```diff
@@ -1161,6 +1265,33 @@ def get_records(self, context: dict | None) -> Iterable[dict | tuple[dict, dict]
         """
         pass
 
+    def get_batches(
```

Comment:
This method should eventually raise an exception if the configured batch encoding is not supported by the tap.